src/lib/api.ts

```diff
@@ -42,80 +66,79 @@ export class AeriesApi {
 		console.log(data);
 
 		if (isFail(data)) {
-			console.log("fail!");
+			console.log("auth fail!");
 			return false;
 		} else {
-			console.log("success");
-			this.token = data.AccessToken;
-			this.student = data.Students[0];
+			console.log("auth success");
+			this.authedStudent.set({
+				Token: data.AccessToken,
+				Student: data.Students[0],
+			});
 			return true;
 		}
 	}
 
-	private genRequest(method: string, url: URL | string, body?: object): Request {
-		let headers: { [id: string] : string} = {};
+	private genRequest(
+		method: string,
+		url: URL | string,
+		body?: object
+	): Request {
+		let headers: { [id: string]: string } = {};
 		if (url !== "/authentication") {
-			if (this.token === null) throw new UninitializedApiError();
-			headers["Authorization"] = `Bearer ${this.token}`;
+			if (this.authedStudent === null) throw new UninitializedApiError();
+			headers["Authorization"] = `Bearer ${get(this.authedStudent)!.Token}`;
 		}
-		body = {url: this.apiUrl + url.toString(), method, headers, ...body};
+		body = { url: this.apiUrl + url.toString(), method, headers, ...body };
 		return new Request("/proxy", {
 			body: JSON.stringify(body),
 			method: "POST",
 		});
 	}
+
 	public async getHomePage(): Promise<HomeScreenData> {
-		if (this.student === null) throw new UninitializedApiError();
+		if (this.authedStudent === null) throw new UninitializedApiError();
 		let resp = await fetch(
 			this.genRequest(
-				"GET", `/student/${this.student.Demographics.StudentID}/homescreendata`
+				"GET",
+				`/student/${
+					get(this.authedStudent)!.Student.Demographics.StudentID
+				}/homescreendata`
 			)
 		);
 		return await resp.json();
 	}
-	
+
 	public async getClassSummaries(): Promise<ClassSummary[]> {
-		if (this.student === null) throw new UninitializedApiError();
+		if (!this.isInitialized()) throw new UninitializedApiError();
 		let resp = await fetch(
 			this.genRequest(
-				"GET", `/student/${this.student.Demographics.StudentID}/classsummary`
+				"GET",
+				`/student/${
+					get(this.authedStudent)!.Student.Demographics.StudentID
+				}/classsummary`
 			)
 		);
-		let datum = await resp.json() as ClassSummaryDatum[];
+		let datum = (await resp.json()) as ClassSummaryDatum[];
 
-		return datum.map(datum => datum.ClassSummary.filter(x => x.Term == "Current Terms")).flat();
+		return datum
+			.map((datum) =>
+				datum.ClassSummary.filter((x) => x.Term == "Current Terms")
```

Comment:
I KNOW I WROTE THIS, but perhaps we could make it so `getClassSummaries()` takes a term, and defaults to the current term. e.g. `public async getClassSummaries(term: string = "Current Terms")`